ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java

```diff
@@ -432,11 +453,18 @@ private boolean isHostGroupBasedDeployment() {
     private void calculateTotalAgentsInHostgroups() {
         int agentsToExpect = 0;
         for (AmbariHostGroup hostGroup : getHostGroups()) {
-                agentsToExpect += hostGroup.getConfig(AmbariHostGroup.INITIAL_SIZE);
+            agentsToExpect += hostGroup.getConfig(AmbariHostGroup.INITIAL_SIZE);
+        }
+        if (getServer().agentOnServer()) {
```

Comment:
I would use the `getAmbariServers()` and loop over to make sure to have all agents.

BTW, the `getServer()` method can be removed.